packages/contracts-bedrock/scripts/upgrades/v1.8.0-permissioned/scripts/bundle.sh

```diff
@@ -0,0 +1,672 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Grab the script directory
+SCRIPT_DIR=$(dirname "$0")
+
+# Load common.sh
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR/common.sh"
+
+# Check required environment variables
+reqenv "NETWORK"
+reqenv "STORAGE_SETTER"
+reqenv "DISPUTE_GAME_FACTORY_PROXY"
+reqenv "DEPLOYMENTS_JSON_PATH"
+
+# Load addresses from deployments json
+PROXY_ADMIN=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "ProxyAdmin")
+OPTIMISM_PORTAL_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "OptimismPortalProxy")
+SYSTEM_CONFIG_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "SystemConfigProxy")
+L1_CROSS_DOMAIN_MESSENGER_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "L1CrossDomainMessengerProxy" "Proxy__OVM_L1CrossDomainMessenger")
+L1_STANDARD_BRIDGE_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "L1StandardBridgeProxy" "Proxy__OVM_L1StandardBridge")
+L1_ERC721_BRIDGE_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "L1ERC721BridgeProxy")
+OPTIMISM_MINTABLE_ERC20_FACTORY_PROXY=$(load_local_address "$DEPLOYMENTS_JSON_PATH" "OptimismMintableERC20FactoryProxy")
+
+# Fetch addresses from standard address toml
+SYSTEM_CONFIG_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "system_config")
+OPTIMISM_PORTAL_2_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "optimism_portal")
+L1_CROSS_DOMAIN_MESSENGER_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "l1_cross_domain_messenger")
+L1_STANDARD_BRIDGE_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "l1_standard_bridge")
+L1_ERC721_BRIDGE_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "l1_erc721_bridge")
+OPTIMISM_MINTABLE_ERC20_FACTORY_IMPL=$(fetch_standard_address "$NETWORK" "$CONTRACTS_VERSION" "optimism_mintable_erc20_factory")
+
+# Fetch SuperchainConfigProxy address
+SUPERCHAIN_CONFIG_PROXY=$(fetch_superchain_config_address "$NETWORK")
+
+# We need to re-generate the SystemConfig initialization call
+# We want to use the exact same values that the SystemConfig is already using
+SYSTEM_CONFIG_OWNER=$(cast call "$SYSTEM_CONFIG_PROXY" "owner()")
+SYSTEM_CONFIG_BASE_FEE_SCALAR=$(cast call "$SYSTEM_CONFIG_PROXY" "basefeeScalar()")
```

Comment:
This is where the bug is. I believe that you need to call `scalar()` and then assert its on version 1 (most significant byte is `0x01`) then decode it with the inverse of https://github.com/ethereum-optimism/optimism/blob/34d5f66ade24bd1f3ce4ce7c0a6cfc1a6540eca1/packages/contracts-bedrock/src/L1/SystemConfig.sol#L406, the logic is also [here](https://github.com/ethereum-optimism/optimism/tree/develop/op-chain-ops/cmd/ecotone-scalar)

Reply:
```bash
$ cast call 0xa6b72407e2dc9EBF84b839B69A24C88929cf20F7 "basefeeScalar()"
0x0000000000000000000000000000000000000000000000000000000000000000
```

Can confirm it does return 0

```bash
$ cast call 0xa6b72407e2dc9EBF84b839B69A24C88929cf20F7 "scalar()"
0x010000000000000000000000000000000000000000000000000d273000001db0
```

this does not return 0, this is the scalar value

Reply:
This is a consistency problem due to new storage slots being introduced for the basefee scalar rather than it being a function that modifies the single source of truth of the scalar

see: https://github.com/ethereum-optimism/optimism/blob/34d5f66ade24bd1f3ce4ce7c0a6cfc1a6540eca1/packages/contracts-bedrock/src/L1/SystemConfig.sol#L403-L404

This probably should have been implemented in a way that didn't add new storage slots to prevent this consistency issue

Reply:
I've removed the re-initialization of SystemConfig from this task as it assumes that the chain is already running `op-contracts/v1.8.0` and the StorageSetter is already being used to zero out the L2OO address and set the DisputeGameFactory address so the initialisation would be a no-op apart from this scalar issue.

How to handle the scalar is being tracked in https://github.com/ethereum-optimism/superchain-ops/issues/394 but that is part of the upgrade to op-contracts/v1.8.0 and this runbook is starting from already running 1.8.0 (just in L2OO config but the SystemConfig should already have been upgraded).